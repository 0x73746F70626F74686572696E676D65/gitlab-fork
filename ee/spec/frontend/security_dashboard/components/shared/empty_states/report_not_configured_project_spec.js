import { GlButton } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import ReportNotConfiguredProject from 'ee/security_dashboard/components/shared/empty_states/report_not_configured_project.vue';

describe('Project report not configured component', () => {
  let wrapper;
  const emptyStateSvgPath = '/placeholder.svg';
  const securityConfigurationPath = '/configuration';
  const securityDashboardHelpPath = '/help';
  const newVulnerabilityPath = '/vulnerability/new';

  const findButton = () => wrapper.findComponent(GlButton);

  const createComponent = ({ provide } = {}) => {
    wrapper = shallowMount(ReportNotConfiguredProject, {
      provide: {
        emptyStateSvgPath,
        securityConfigurationPath,
        securityDashboardHelpPath,
        newVulnerabilityPath,
        ...provide,
      },
    });
  };

  describe.each`
    provide                                                      | expectedShow
    ${{ newVulnerabilityPath: '', canAdminVulnerability: true }} | ${false}
    ${{ newVulnerabilityPath, canAdminVulnerability: false }}    | ${false}
    ${{ newVulnerabilityPath, canAdminVulnerability: true }}     | ${true}
  `('should display or hide the button based on the condition', ({ provide, expectedShow }) => {
    beforeEach(() => {
      createComponent({ provide });
    });

    it(`shows the button: ${expectedShow}`, () => {
      expect(findButton().exists()).toBe(expectedShow);
    });
  });

  describe('when button is shown', () => {
    beforeEach(() => {
      createComponent({
        provide: {
          canAdminVulnerability: true,
        },
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
