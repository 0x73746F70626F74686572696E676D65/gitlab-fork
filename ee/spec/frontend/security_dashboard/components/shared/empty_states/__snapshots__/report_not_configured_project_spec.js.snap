// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project report not configured component when button is shown matches the snapshot 1`] = `
"<div>
  <div class=\\"gl-my-4 gl-text-right\\">
    <gl-button-stub category=\\"primary\\" variant=\\"default\\" size=\\"medium\\" icon=\\"plus\\" buttontextclasses=\\"\\" href=\\"/vulnerability/new\\" class=\\"gl-ml-auto\\">
      Submit vulnerability
    </gl-button-stub>
  </div>
  <gl-empty-state-stub title=\\"Monitor vulnerabilities in your project\\" svgpath=\\"/placeholder.svg\\" description=\\"Manage and track vulnerabilities identified in your project. Vulnerabilities are shown here when security testing is configured.\\" primarybuttonlink=\\"/configuration\\" primarybuttontext=\\"Configure security testing\\" secondarybuttonlink=\\"/help\\" secondarybuttontext=\\"Learn more\\" invertindarkmode=\\"true\\"></gl-empty-state-stub>
</div>"
`;
